package app.twallet.air.walletcore.models.blockchain

import app.twallet.air.walletcontext.models.MBlockchainNetwork

interface MBlockchainConfig {
    val gas: MBlockchain.Gas?
    val nativeIcon: Int?
        get() = null
    val symbolIcon: Int?
    val symbolIconPadded: Int?
    val receiveOrnamentImage: Int?
    val qrIcon: Int?
    val qrGradientColors: IntArray?

    val displayColor: Int
    val feeCheckAddress: String?
    val isCommentSupported: Boolean
    val isEncryptedCommentSupported: Boolean
    val burnAddress: String?
    val isOnchainSwapSupported: Boolean get() = false
    val canSwapByBuyAmount: Boolean get() = false
    val multiWalletSupport: MultiWalletSupport?

    val chainStandard: String? get() = null
    val defaultDerivationPath: String? get() = null
    val walletConnectChainIds: Map<MBlockchainNetwork, Int> get() = emptyMap()

    fun isValidAddress(address: String): Boolean
    fun isValidDNS(address: String): Boolean = false
    fun idToTxHash(id: String?): String? = null

    fun transactionExplorers(): List<MBlockchainExplorer>
    fun addressExplorers(): List<MBlockchainExplorer>
    fun tokenExplorer(): MBlockchainExplorer?
    fun nftExplorer(): MBlockchainExplorer?
}
